Mathlib/Analysis/Convex/Radon.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2023 Vasily Nesterov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Vasily Nesterov
+-/
+import Mathlib.Analysis.Convex.Combination
+import Mathlib.Tactic.Linarith
+
+/-!
+# Radon's theorem on convex sets
+
+Radon's theorem states that any affine dependent set can be partitioned into two sets whose convex
+hulls intersect.
+
+## Tags
+
+convex hull, radon, affine independence
+-/
+
+open Finset Set
+open BigOperators
+
+variable {ι 𝕜 E : Type*} [LinearOrderedField 𝕜] [AddCommGroup E] [Module 𝕜 E] {f : ι → E}
+
+/-- **Radon theorem on convex sets**: Any family `f` of affine dependent vectors contains a set `I`
+with the property that convex hulls of `I` and `Iᶜ` intersect. -/
+theorem radon_partition (h : ¬ AffineIndependent 𝕜 f) :
+    ∃ I, (convexHull 𝕜 (f '' I) ∩ convexHull 𝕜 (f '' Iᶜ)).Nonempty := by
```

Comment:
@YaelDillies, do you think using `Disjoint` here would be better or worse?

Reply:
Probably worse. That's neither how we prove nor use it.